File: LeetCode/1684. Count the Number of Consistent Strings/Solution.java
/**
 * Runtime: 42 ms
 * Memory Usage: 38.4 MB 
 */
class Solution {
    public int toCode(String s) {
        return s.chars()
            .boxed()
            .map(x -> x - 'a')
            .reduce(0, (x, y) -> x | (1 << y));
    }
    public int countConsistentStrings(String allowed, String[] words) {
        int allowedCode = toCode(allowed);
        int ans = 0;
        for (String word: words) {
            ans += ((allowedCode | toCode(word)) == allowedCode) ? 1 : 0;
        }
        return ans;
    }
}

File: LeetCode/1684. Count the Number of Consistent Strings/solution.py
"""
Runtime: 200 ms
Memory Usage: 16.7 MB
"""
from functools import reduce
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        tocode = lambda s: reduce(lambda x, y: x | (1 << (ord(y) - ord('a'))), s, 0)
        allowed_code = tocode(allowed)
        ans = 0
        for word in words:
            ans += 1 if (allowed_code | tocode(word)) == allowed_code else 0
        return ans
